cip.go

```diff
@@ -108,38 +112,68 @@ func main() {
 	var mfest reg.Manifest
 	var srcRegistry *reg.RegistryContext
 	var err error
+	var mfests []reg.Manifest
+	sc := reg.SyncContext{}
+	mi := make(reg.MasterInventory)
+
 	if len(*snapshotPtr) > 0 {
 		srcRegistry = &reg.RegistryContext{
 			Name:           reg.RegistryName(*snapshotPtr),
 			ServiceAccount: *snapshotSvcAccPtr,
 			Src:            true,
 		}
-		mfest = reg.Manifest{
-			Registries: []reg.RegistryContext{
-				*srcRegistry,
+		mfests = []reg.Manifest{
+			{
+				Registries: []reg.RegistryContext{
+					*srcRegistry,
+				},
+				Images: []reg.Image{},
 			},
-
-			Images: []reg.Image{},
 		}
 	} else {
-		if *manifestPtr == "" {
-			klog.Fatal(fmt.Errorf("-manifest=... flag is required"))
+		if *manifestPtr == "" && *manifestDirPtr == "" {
+			klog.Fatal(fmt.Errorf("-manifest=... or -manifestDir=... flag is required"))
 		}
+	}
+
+	if *manifestPtr != "" {
 		mfest, err = reg.ParseManifestFromFile(*manifestPtr)
 		if err != nil {
 			klog.Fatal(err)
 		}
-	}
+		mfests = append(mfests, mfest)
+		for _, registry := range mfest.Registries {
```

Comment:
Not clear why build sc from mfests is different in the file vs directory case, and also why we don't need to pre-allocate mi in the directory case.

Reply:
Oh I suppose they could share the same sc (I think you are asking why we are calling MakeSyncContext twice?)